### DISCUSSION TITLE
Documentation for using the API? Adding metadata to test runs

### DISCUSSION THREAD

**Discussion by BrianBaggs** [test-results-reporter/reporter#247] [Q&A] [answered]
Hello, I just started using testbeats and everytime I send results, in the portal it says Ran by Unknown and there is hardly any metadata. I saw a screenshot with someone who had a bunch of extra data listed but I have no idea how to add this data, so if anyone could help it would be greatly appreciated. Stuff like my name, and changing the name of the test run because it always ends up as demo-run. I poured over your documentation and searched the discussions and could not find anything to help.

**Comment by ASaiAnudeep** [comment]
Hello @BrianBaggs , welcome to testbeats.

> Test Run MetaData

I understand you're seeing "Ran by Unknown" and limited metadata in the portal. This typically happens when metadata isn’t captured from your CI system. If you're using a CI tool like GitHub Actions, the metadata should be automatically captured.

Could you let us know where you’re running your tests? That way, we can provide more specific guidance. You can also find the source code here: [Capture MetaData](https://github.com/test-results-reporter/testbeats/blob/main/src/helpers/ci.js)

>  Custom Test Run Title

Regarding the test run name, you can customize it using the `title` property. You can find more details here: [TestBeats Documentation](https://docs.testbeats.com/targets/slack.html).

We appreciate your feedback and will work on improving the documentation!

**Reply by BrianBaggs** [reply]
Hello, thanks for the quick reply! 

I am running the tests locally using Playwright, and then uploading the Junit XML file using your command line tool, but it seems to say ran by Unknown every time in the testbeats portal. Also, I tried to change the name of the run using --run 'October 9th' but that did not change anything, it still said demo-run. See below screenshot.
![Ran by unknown](https://github.com/user-attachments/assets/fb7b611c-a66f-4bd1-9163-bbcc7b7f3277)

**Reply by ASaiAnudeep** [reply, marked as answer]
Understood! I’ll work on capturing the metadata from local runs as part of this [ticket](https://github.com/test-results-reporter/testbeats/issues/225).

> Also, I tried to change the name of the run using --run 'October 9th' but that did not change anything, it still said demo-run.

Could you please share the full command you’re using locally? That will help us troubleshoot the issue.

**Reply by BrianBaggs** [reply]
Sure! The command was - `testbeats --project 'Regression Testing' --run 'Oct9'` All of the info about the API key and the Junit thing is in the config file, which I used your documentation to create and as far as I can tell it works fine except for the issue mentioned above. By the way, I have an alias created for the testbeats command that replaces it with this - `npx testbeats@latest publish -c testbeats.config.json` so that part gets appended to the rest of the command. Also, I run it right from my testbeats folder, where I send the XML results and have the testbeats.config.json file.

**Reply by BrianBaggs** [reply]
Also, thank you for your help, adding that feature will help me immensely. I'm not sure if it's going to be a matter of adding something like --name 'Brian' to the command, or adding it to the config file or XML results. Here is an image of the top of the run in testbeats. As you can see, there isn't much info there.
![Top of testbeats run](https://github.com/user-attachments/assets/50a47c30-b590-48a5-9af1-d862036c3edd)

**Reply by ASaiAnudeep** [reply]
Created an issue to track the work - https://github.com/test-results-reporter/testbeats/issues/250

For the time being please mention the run name in the `config.json` file

**Reply by ASaiAnudeep** [reply]
This has been fixed in the latest release [v2.1.5](https://github.com/test-results-reporter/testbeats/releases/tag/v2.1.5)
